frontend/src/metabase/setup/selectors.ts

```diff
@@ -77,19 +79,29 @@ export const getAvailableLocales = (state: State): LocaleData[] => {
 export const getIsEmailConfigured = (state: State): boolean => {
   return getSetting(state, "email-configured?");
 };
-
+``;
 export const getSteps = (state: State) => {
   const usageReason = getUsageReason(state);
   const activeStep = getStep(state);
 
+  // TODO: avoid using MetabaseSettings in a selector
+  // we can't do it right now as this selector is used in the reducer that doesn't have access to the whole state
+  const tokenFeatures = MetabaseSettings.get("token-features");
```

Comment:
It's uncommon for reducers to know about the whole state other than its own slice. The way this problem is solved is via sending necessary information via action payload.

You could do this by sending necessary data e.g. `isPaidPlan` in [the action](https://github.com/metabase/metabase/pull/38953/files#diff-f0f1344935a4c128194840b010588c50a1260aa9dae823a4b8c51dee7e130284R97). You can access `getState` in thunk action and that's how you could get `token-features` setting from the global Redux state. Then probably either send `isPaidPlan` or `shouldShowLicenseStep` depending on how you think it's best to encapsulate the information.

After that this reducer won't need the access to `MetabaseSettings` anymore.